ynr/apps/bulk_adding/helpers.py

```diff
@@ -69,3 +75,129 @@ def update_person(
         popit_person_new_version=change_metadata["version_id"],
         source=change_metadata["information_source"],
     )
+
+
+class CSVImporter:
+    def __init__(self, file_name, election):
+        self.csv_data = list(csv.DictReader(open(file_name).readlines()))
+        self.header_rows = self.csv_data[0].keys()
+        self.election = election
+
+        self.validate_posts()
+
+        self.ballots = {}
+        for ballot in self.election.postextraelection_set.all():
+            self.ballots[ballot.ballot_paper_id] = {
+                "ballot": ballot,
+                "data": [],
+            }
+
+    @property
+    def post_header_name(self):
+        POST_LABELS = ["Electoral Area Name", "Election Area"]
+        for label in POST_LABELS:
+            if label in self.header_rows:
+                return label
+
+    @property
+    def party_description_header_name(self):
+        DESCRIPTION_FIELDS = ["Description", "Candidates Description"]
+        for description_field in DESCRIPTION_FIELDS:
+            if description_field in self.header_rows:
+                return description_field
+
+    def match_division_to_ballot(self, row):
+        post_name = row[self.post_header_name]
+        for ballot in self.election.postextraelection_set.all():
+            if ballot.post.label == post_name:
+                return ballot
```

Comment:
There are several functions here which only return a value conditionally. For example, if we fail to match the post name to any ballot, this function will implicitly return `None`, but then we're just rolling with it expecting it to be a sensible value. How about we throw an exception at the end of this function if we can't match anything? (and similar comments for `post_header_name()`, `party_description_header_name()` etc). If you're relying on catching the `None` return value, its conventional to explicitly `return None`

Reply:
Also worth noting here is this function is probably going to benefit at some point from the ability to abstract minor differences (e.g: "This Place and That Place" vs "This Place & That Place", "St. Mark's" vs "St Marks" etc), but starting with a strict implementation and loosening in the right ways as we hit examples is sensible :+1:

Reply:
Yes, good points. I've raised some `ValueErrors` in this command now.

On the other matching operations, I think a refactor is in order, as I'm also doing a load of matching in the `resultsbot` app, including things like printing an options list to the command line for manual selecting. 

Pulling that logic out in to a more general "guess things" library would be useful, but I didn't want to get in to that in this PR, or until I've got some actual CSVs to work with.

Reply:
A fuzzy matching library with assumptions for UK place names would be handy in other places (e.g: boundary bot, matching shape files from LGBCE to Electoral Change Orders, etc), but this is definitely a yak shave for another time